astropy/units/decorators.py

```diff
@@ -204,6 +250,18 @@ def wrapper(*func_args, **func_kwargs):
                     targets = param.annotation
                     is_annotation = True
 
+                    # parses to unit if it's an annotation, else return targets
+                    targets = _parse_annotation(targets)
+
+                    # still need to check if its a Union of things, in which
```

Comment:
Could this check be moved to `_parse_annotation` as well?

Reply:
Yes. 👍 Done.